app/policies/user_policy.rb

```diff
@@ -11,8 +11,12 @@ def create?
     is_admin? || is_himself?
   end
 
+  def update?
+    is_himself?
+  end
+
   def is_himself?
-    current_user == record
+    record == current_user
```

Comment:
@vitortalaia @dmitryrck the crazier it seems, it does make sense to change the order, here's the reason:

`ActiveRecord::Base#==`  method tests for: 1) the equality of the classes of the compared objects and 2) their ids. So if you call `record == current_user`, the `record` (an instance of `User`) will see that the class of `current_user` is `UserPresenter` and return that they're not equal. From the other side, if you do `current_user == record` it'll delegate the `==` method to the underneath instance of `User` inside `current_user` (because `current_user` is an instance of `UserPresenter`, which is a `SimpleDelegator`) and then it'll see that their classes are equal and then check for their ids (that are also equal) and return true.